packages/lesswrong/lib/editor/make_editable.tsx

```diff
@@ -37,6 +42,22 @@ export interface MakeEditableOptions {
   hidden?: boolean,
 }
 
+const isSharable = (document: any) : document is SharableDocument => {
+  return "coauthorStatuses" in document || "shareWithUsers" in document || "sharingSettings" in document
+}
+
+export const getOriginalContents = (user: DbUser|null, doc: DbObject, originalContents: EditableFieldContents["originalContents"]) => {
+  let canViewOriginalContents: () => boolean
+  if (isSharable(doc)) {
+    canViewOriginalContents = () => userIsSharedOn(user, doc)
+  } else {
+    canViewOriginalContents = () => true
+  }
```

Comment:
```suggestion
  const canViewOriginalContents = (user: DbUser|null, doc: DbObject) => isSharable(doc) ? userIsSharedOn(user, doc) : true
```
I think this would be a bit clearer. In general I think it's better to pass args directly into an inline function because then it isn't coupled as strongly to the scope it was defined in. E.g. we are relying on the `user` in `userCanReadField(user...` being the same one as in `userIsSharedOn(user, doc)`, in this case that obviously is true, but you can imagine a case where the function gets passed around a bit and we lose track of that